docs/pages/blog/making-customizable-components.md

```diff
@@ -0,0 +1,287 @@
+---
+title: 'Making customizable components'
+description: The use case of the data grid
+date: 2022-06-23T00:00:00.000Z
+authors: ['alexfauquette']
+tags: ['MUI X']
+---
+
+At MUI, our goal is to empower as many people as possible to build great UIs.
+To do so, we build React components and documentation used by hundreds of thousands of developers on everything from minor side projects to massive company websites.
+
+This variety of users presents a dilemma: hobbyists working on side projects want fully built components that work right out of the box, so they can focus on the application logic; many larger companies, by contrast, want to be able to fully customize components to respect their brand design.
+Managing these contradictory needs becomes harder as component complexity increases.
+
+This article reviews several different approaches to customizing components, as well as the various tradeoffs associated with each method.
+Along the way, we’ll explore how these tradeoffs ultimately led to the solution that we’ve settled on for customizing MUI components.
+
+## Style modification
+
+If you already use a styling library, don’t hesitate to skip this part.
+See you back in the [Logic modification section](#logic-modification)
+
+### Good old CSS
+
+Let's start with the easiest part: modifying the style.
+This will necessarily involve CSS.
+Especially the notion of [specificity](https://developer.mozilla.org/en-US/docs/Web/CSS/Specificity), which says that if an element is targeted by two CSS selectors, the browser will apply the more specific one.
+Most of the time it is the selector using more classes.
+
+For example, if we look at a switch component, we have multiple subcomponents that we could expect to modify.
+For each of them, we assign a specific CSS class:
+
+<img src="/static/blog/making-customizable-components/switchHighlighted.png" style="width: 796px; margin-top: 16px; margin-bottom: 16px;" alt="Switch component with highlighted sub components" />
```

Comment:
Are we sure about the `width` value? The scale isn't 1:1 it might fee a bit weird. 

```suggestion
<img src="/static/blog/making-customizable-components/switchHighlighted.png" style="width: 796px; margin-top: 16px; margin-bottom: 16px;" alt="Switch component with highlighted sub components" />
```

As a side note, it would have been great to have an image with a pixel-to-device resolution of 2 😁 but we can ignore this, it only impact 1/3 of our audience, it's biased toward designers.